src/main/java/tp/cap5buddy/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,174 @@
+        return new CommandResult(message, isExit());
+    }
+
+    /**
+     * Creates edited version of the original task.
+     *
+     * @param taskToEdit original task to be edited.
+     * @return the edited task.
+     * */
+    public Task createEditedTask(Task taskToEdit, EditTaskDescriptor editTaskDescriptor) {
+        assert taskToEdit != null;
+
+        Type updatedType = editTaskDescriptor.getType().orElse(taskToEdit.getType());
+        Description updatedDescription = editTaskDescriptor.getDescription().orElse(taskToEdit.getDescription());
+        Priority updatedPriority = editTaskDescriptor.getPriority().orElse(taskToEdit.getPriority());
+        Date updatedDate = editTaskDescriptor.getDate().orElse(taskToEdit.getDate());
+        Status status = taskToEdit.getStatus();
+
+        return new Task(updatedType, updatedDescription, updatedPriority, updatedDate, status);
+    }
+
+    /**
+     * Creates success message after successfully executing this command.
+     *
+     * @param taskToEdit the original task.
+     * @param editedTask the task after editing.
+     * @return String that represent a success message followed by the content of the original
+     *         task and the content after the task has been edited.
+     */
+    public String createSuccessMessage(Task taskToEdit, Task editedTask) {
+        String message = ""
+            + "Task " + (indexToEdit + 1) + " has been successfully edited!\n\n"
+            + "Before edited: \n"
+            + "Type        : " + taskToEdit.getType() + "\n"
+            + "Description : " + taskToEdit.getDescription() + "\n"
+            + "Date        : " + taskToEdit.getDate() + "\n"
+            + "Priority    : " + taskToEdit.getPriority() + "\n\n"
+            + "After Edited: \n"
+            + "Type        : " + editedTask.getType() + "\n"
+            + "Description : " + editedTask.getDescription() + "\n"
+            + "Date        : " + editedTask.getDate() + "\n"
+            + "Priority    : " + editedTask.getPriority() + "\n";
+        return message;
+    }
+
+    /**
+     * Indicates if the application session has ended.
+     *
+     * @return True if the session has been terminated, false otherwise.
+     */
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Stores the details to edit the task with. Each non-empty field value will replace the
+     * corresponding field value of the task.
+     */
+    public static class EditTaskDescriptor {
+        /**
+         * the edited type.
+         */
+        private Type type;
+        /**
+         * the edited description.
+         */
+        private Description description;
+        /**
+         * the edited date/deadline.
+         */
+        private Date date;
+        /**
+         * the edited priority.
+         */
+        private Priority priority;
+
+        /**
+         * Constructs an empty EditTaskDescriptor.
+         */
```

Comment:
We could perhaps consider shifting out EditTaskDescriptor as a separate class rather than a nested class.

Reply:
Yes, I think that's a good idea